Mathlib/FieldTheory/Galois/Infinite.lean

```diff
@@ -0,0 +1,282 @@
+    h y (IntermediateField.le_normalClosure L hy))) d)
+  have fix : ∀ x ∈ IntermediateField.fixedField H.toSubgroup ⊓ ↑L', σ x = x :=
+    fun x hx ↦ ((mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mp hσ) x hx.1
+  rw [restrict_fixedField H.1 L'.1] at fix
+  have : (restrictNormalHom L') σ ∈ (Subgroup.map (restrictNormalHom L') H.1) := by
+    rw [← IntermediateField.fixingSubgroup_fixedField (Subgroup.map (restrictNormalHom L') H.1)]
+    apply (mem_fixingSubgroup_iff (L' ≃ₐ[k] L')).mpr
+    intro y hy
+    apply Subtype.val_injective
+    simp only [AlgEquiv.smul_def, restrictNormalHom_apply L'.1 σ y,
+      fix y.1 ((IntermediateField.mem_lift y).mpr hy)]
+  rcases this with ⟨h, mem, eq⟩
+  have : h ∈ σ • L'.1.fixingSubgroup.carrier := by
+    use σ⁻¹ * h
+    simp only [Subsemigroup.mem_carrier, Submonoid.mem_toSubsemigroup, Subgroup.mem_toSubmonoid,
+      smul_eq_mul, mul_inv_cancel_left, and_true]
+    apply (mem_fixingSubgroup_iff (K ≃ₐ[k] K)).mpr
+    intro y hy
+    simp only [AlgEquiv.smul_def, AlgEquiv.mul_apply]
+    have : ((restrictNormalHom L') h ⟨y,hy⟩).1 = ((restrictNormalHom L') σ ⟨y,hy⟩).1 := by rw [eq]
+    rw [restrictNormalHom_apply L'.1 h ⟨y, hy⟩, restrictNormalHom_apply L'.1 σ ⟨y, hy⟩] at this
+    simp only [this, ← AlgEquiv.mul_apply, inv_mul_cancel, one_apply]
+  absurd compl
+  apply Set.not_subset.mpr
+  use h
+  simpa only [this, Set.mem_compl_iff, Subsemigroup.mem_carrier, Submonoid.mem_toSubsemigroup,
+    Subgroup.mem_toSubmonoid, not_not, true_and] using mem
+
+/-- The Galois correspondence from intermediate fields to ClosedSubgroup. -/
+def IntermediateFieldEquivClosedSubgroup [IsGalois k K] :
+    IntermediateField k K ≃o (ClosedSubgroup (K ≃ₐ[k] K))ᵒᵈ where
+  toFun := fun L =>
+    { L.fixingSubgroup with
+      isClosed' := fixingSubgroup_isClosed L }
+  invFun := fun H => IntermediateField.fixedField H.1
+  left_inv := fun L => fixedField_fixingSubgroup L
+  right_inv := by
+    intro H
+    simp_rw [fixingSubgroup_fixedField H]
+    rfl
+  map_rel_iff' := by
+    intro L₁ L₂
+    show L₁.fixingSubgroup ≥ L₂.fixingSubgroup ↔ L₁ ≤ L₂
+    rw [← fixedField_fixingSubgroup L₂, IntermediateField.le_iff_le, fixedField_fixingSubgroup L₂]
+
+/-- The Galois correspondence as a `GaloisInsertion` -/
+def GaloisInsertionIntermediateFieldSubgroup [IsGalois k K] :
+    GaloisInsertion (OrderDual.toDual ∘ fun (E : IntermediateField k K) ↦
+      (⟨E.fixingSubgroup, fixingSubgroup_isClosed E⟩ : ClosedSubgroup (K ≃ₐ[k] K)))
+      ((fun (H : ClosedSubgroup (K ≃ₐ[k] K)) ↦ IntermediateField.fixedField H) ∘
+        OrderDual.toDual) where
+  choice E _ := ⟨E.fixingSubgroup, fixingSubgroup_isClosed E⟩
+  gc E H := (IntermediateField.le_iff_le H.1 E).symm
+  le_l_u H := by
+    simp only [Function.comp_apply, fixingSubgroup_fixedField]
+    rfl
+  choice_eq _ _ := rfl
+
+/-- The Galois correspondence as a `GaloisCoinsertion` -/
+def GaloisCoinsertionIntermediateFieldSubgroup [IsGalois k K] :
```

Comment:
This is `OrderIso.toGaloisCoinsertion` I think?

Reply:
I found out that `galoisCoinsertionIntermediateFieldSubgroup` in the `Galois.Basic` can actually prove by this.

Reply:
Let's get this one over the line first :-)